src/metric_config.hpp

```diff
@@ -30,73 +32,64 @@ class MetricConfig {
     bool metricsEnabled;
     std::string endpointsPath;
 
-    // Request Success/Fail
-    // TFS
-    bool requestSuccessGrpcPredict;
-    bool requestSuccessGrpcGetModelMetadata;
-    bool requestSuccessGrpcGetModelStatus;
-
-    bool requestSuccessRestPredict;
-    bool requestSuccessRestGetModelMetadata;
-    bool requestSuccessRestGetModelStatus;
-
-    bool requestFailGrpcPredict;
-    bool requestFailGrpcGetModelMetadata;
-    bool requestFailGrpcGetModelStatus;
-
-    bool requestFailRestPredict;
-    bool requestFailRestGetModelMetadata;
-    bool requestFailRestGetModelStatus;
-
-    // KFS
-    bool requestSuccessGrpcModelInfer;
-    bool requestSuccessGrpcModelMetadata;
-    bool requestSuccessGrpcModelReady;
-
-    bool requestSuccessRestModelInfer;
-    bool requestSuccessRestModelMetadata;
-    bool requestSuccessRestModelReady;
-
-    bool requestFailGrpcModelInfer;
-    bool requestFailGrpcModelMetadata;
-    bool requestFailGrpcModelReady;
-
-    bool requestFailRestModelInfer;
-    bool requestFailRestModelMetadata;
-    bool requestFailRestModelReady;
-
-    bool requestTimeGrpc;
-    bool requestTimeRest;
-
-    bool inferenceTime;
-    bool waitForInferReqTime;
-
-    bool streams;
-
-    bool inferReqQueueSize;
-    bool inferReqActive;
-
-    bool currentRequests;
+    std::unordered_set<std::string> enabledFamiliesList;
+
+    std::unordered_map<std::string, std::unordered_set<std::string>> additionalMetricFamilies = {
+        {"ovms_streams", {"ovms_streams"}},
+        {"ovms_infer_req_queue_size", {"ovms_infer_req_queue_size"}},
+        {"ovms_infer_req_active", {"ovms_infer_req_active"}},
+        {"ovms_current_requests", {"ovms_current_requests"}}};
+
+    std::unordered_map<std::string, std::unordered_set<std::string>> defaultMetricFamilies = {
+        {"ovms_requests_success",
+            {"ovms_requests_success_grpc_predict",
+                "ovms_requests_success_grpc_getmodelmetadata",
+                "ovms_requests_success_grpc_getmodelstatus",
+                "ovms_requests_success_grpc_modelinfer",
+                "ovms_requests_success_grpc_modelmetadata",
+                "ovms_requests_success_grpc_modelready",
+                "ovms_requests_success_rest_modelinfer",
+                "ovms_requests_success_rest_predict",
+                "ovms_requests_success_rest_get_modelmetadata",
+                "ovms_requests_success_rest_get_modelstatus",
+                "ovms_requests_success_rest_modelmetadata",
+                "ovms_requests_success_rest_modelready"}},
+        {"ovms_requests_fail",
+            {"ovms_requests_fail_grpc_predict",
+                "ovms_requests_fail_grpc_getmodelmetadata",
```

Comment:
Why do we have those `"ovms_requests_fail_grpc_getmodelmetadata",...` metrics? We never use it in model metric reporter as far I can see.

Reply:
We can add it in the future when we need it. The format of those single metric filters will differ anyway (will not be a single string but json object instead)

Reply:
Ok